go/integTest/shared_commands_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package integTest
+
+import (
+	"time"
+
+	"github.com/aws/glide-for-redis/go/glide/api"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	keyName      = "key"
+	initialValue = "value"
+	anotherValue = "value2"
+)
+
+func (suite *GlideTestSuite) TestSetAndGet_noOptions() {
+	suite.runWithDefaultClients(func(client api.BaseClient) {
```

Comment:
You need to do renamings in `glide_test_suite_test.go` too, but `extractRedisVersion` should support both redis and valkey (see IT of other clients)

Reply:
Referred the IT of Csharp and modified the extractRedisVersion function.